platform/ios/src/MGLMapView.mm

```diff
@@ -4683,6 +4676,47 @@ - (void)updateAnnotationViews
         if (annotationView)
         {
             annotationView.center = [self convertCoordinate:annotationContext.annotation.coordinate toPointToView:self];
+            // Pin the callout view to the view annotation
+            UIView <MGLCalloutView> *calloutView = self.calloutViewForSelectedAnnotation;
+            if (calloutView && calloutView.representedObject == annotationContext.annotation) {
+                CGPoint point = annotationView.center;
+                point.x += annotationView.centerOffset.dx;
+                point.y += annotationView.centerOffset.dy;
+                point.y -= annotationView.bounds.size.height/2.0f;
+                if ( ! CGPointEqualToPoint(calloutView.center, point))
+                {
+                    calloutView.center = point;
+                }
+            }
+        } else {
+            // Pin the callout view to the gl annotation
+            UIView <MGLCalloutView> *calloutView = self.calloutViewForSelectedAnnotation;
```

Comment:
Like `-selectAnnotation:animated:`, this method should use `-positioningRectForCalloutForAnnotationWithTag:`, which accounts for the padding around the annotation image and avoids half-pixel issues. To convert from the positioning rect to an anchor point, use `CGRectGetMidX()` and `CGRectGetMinY`.